src/App.js

```diff
@@ -1,27 +1,74 @@
 import React, { Component } from 'react';
 import './App.css';
 import GroupList from './GroupList';
+import TodoList from './TodoList';
 
 class App extends Component {
   state = {
-    groupList: [{ name: 'sally' }, { name: 'sun' }],
+    groupList: [
+      {
+        name: 'coding',
+        todoList: [
+          { content: 'todo app', completed: false },
+          { content: 'next.js', completed: false },
+        ],
+      },
+    ],
     selectedIndex: null,
   };
 
   addGroup = (name) => {
+    const { groupList } = this.state;
+
+    this.setState({
+      groupList: [...groupList, { name, todoList: [] }],
+      selectedIndex: groupList.length,
+    });
+  };
+
+  selectGroup = (index) => {
+    this.setState({ selectedIndex: index });
+  };
+
+  renderTodoList = (groupList, selectedIndex, addTodo) => {
+    if (selectedIndex !== null) {
+      const selectedGroup = groupList[selectedIndex];
+      const todoList = selectedGroup['todoList'];
+
+      return <TodoList todoList={todoList} addTodo={addTodo} />;
+    }
+  };
+
+  addTodo = (content) => {
+    const { groupList, selectedIndex } = this.state;
+    const newTodo = { content, completed: false };
+
     this.setState({
-      groupList: [...this.state.groupList, { name }],
+      groupList: groupList.map((group, index) => {
+        if (index === selectedIndex) {
+          group.todoList = [...group.todoList, newTodo];
+        }
+        return group;
+      }),
```

Comment:
```suggestion
      groupList: groupList.map((group, index) => {
        ...group,
        todoList: index === selectedIndex ? [...group.todoList, newTodo] : group.todoList
      }),
```
결과적으로는 setState를 통해 state를 업데이트하고 있기는 하지만, 49번 줄에서 기존 groupList를 수정하는 코드가 있어요.
React에서 state의 직접 수정은 지양하는게 좋으니, 작은 코드라도 수정하는게 좋을 것 같아요.

Reply:
state를 직접 수정하지 않도록 변경하였고 삼항연산자 대신에 if문을 사용하였습니다. (반영+)